### PR TITLE
fix(status-go)_: add missing status-go dependencies

### PR DESCRIPTION
**Pull request by igor-sirotin**
Requires things like: `go-bindata`, `mockgen`, `protobuf`

Fixes:
- https://github.com/status-im/status-desktop/issues/16477

For more details see:
- https://github.com/status-im/status-go/pull/5878

### CODE REVIEW

**Comment by siddarthkay** [comment]
To fix `mockgen` issues on linux I had to first remove the existing old `mockgen` library and then install latest one from these instructions -> https://github.com/uber-go/mock?tab=readme-ov-file#installation

```
> which mockgen
/usr/bin/mockgen

> sudo rm -rf /usr/bin/mockgen
> go install go.uber.org/mock/mockgen@latest
go: downloading go.uber.org/mock v0.4.0
go: downloading golang.org/x/mod v0.11.0
go: downloading golang.org/x/tools v0.2.0
go: downloading golang.org/x/sys v0.1.0

> export PATH=$PATH:$(go env GOPATH)/bin
> mockgen -version
v0.4.0
```
I've done this on all linux hosts.

**Comment by siddarthkay** [comment]
`macos` `arm64` pipeline fails at `package` stage with : 

```
[2024-10-07T16:20:32.588Z] [92mBuilding:[39m bin/nim_status_client
[2024-10-07T16:20:32.588Z] /Users/jenkins/workspace/
s_macos_aarch64_package_PR-16470/vendor/nimbus-build-system/
scripts/env.sh: line 29: cd: /Users/jenkins/workspace/
s_macos_aarch64_package_PR-16470/vendor/nimbus-build-system/
scripts/../../.nimble: No such file or directory
[2024-10-07T16:20:32.589Z] /Users/jenkins/workspace/
s_macos_aarch64_package_PR-16470/vendor/nimbus-build-system/
scripts/env.sh: line 83: nim: command not found
```

The surprising thing however is that there was no change to `nimbus-build-system` in this PR.
Wonder why this failure happens.

File: ci/Dockerfile [hunk 1]
```diff
@@ -89,6 +89,14 @@ RUN GOLANG_SHA256="e330e5d977bf4f3bdc157bc46cf41afa5b13d66c914e12fd6b694ccda65fc
  && rm "${GOLANG_TARBALL}" \
  && sudo ln -s /usr/local/go/bin/go /usr/local/bin
 
+# Install Protobuf
+RUN PROTOC_SHA256="b9415563e9e147b95f5d3573c92f9e25117a1ad0123cd0b6bd5e8590ea26bd7d" \
+ && PROTOC_TARBALL="protoc-gen-go.v1.35.1.linux.amd64.tar.gz" \
```
Comment (siddarthkay) [inline ci/Dockerfile:94]: I think this version is too new for `status-go` 
we could try using `v1.34.1` -> https://github.com/status-im/status-go/blob/55bad8fd7b48439ded3abef549c11509a4388272/go.mod#L71

**Review by siddarthkay — APPROVED** [review]

**Review by dlipicar — APPROVED** [review]
image
but CI builds build, macOS local build builds, gave the macOS build a quick test and it seems to work fine.

**Comment by siddarthkay** [comment]
@status-im/desktop-qa : okay to merge?

**Comment by anastasiyaig** [comment]
first time i see this PR. I can test it from desktop client side with autotests, would that be efficient? @siddarthkay

**Comment by anastasiyaig** [comment]
sorry i misread :D @siddarthkay its fine from me

**Comment by jakubgs** [comment]
Infra changes:

- [`infra-ci#0113b291`](https://github.com/status-im/infra-ci/commit/0113b291) - status-desktop-setup: support status-go generation

**Comment by yakimant** [comment]
Please add me as reviewer next time for such changes (and also all makefile, nix, ci chachges).
Thanks!

Right now I wonder, why local nix shell is running protobuf compilation (which takes a lot on VM).
I will check the nic cache CI job and uploade manyally if needed.

**Comment by jakubgs** [comment]
You're right, sorry, but I did post it in our PRs channel and you did even respond to my message :D.

**Pull request merged**